libs/ibm/langchain_ibm/chat_models.py

```diff
@@ -1225,15 +1235,19 @@ def _generate(
         if self.watsonx_model_gateway is not None:
             call_kwargs = {**kwargs, **updated_params}
             response = self._call_model_gateway(
+                self.watsonx_model_gateway,
                 model=self.model,
                 messages=message_dicts,
                 **call_kwargs,
             )
-        else:
+        elif self.watsonx_model is not None:
             response = self.watsonx_model.chat(
                 messages=message_dicts,
                 **(kwargs | {"params": updated_params}),
             )
+        else:
+            msg = "Should never reach here"
```

Comment:
By construction, either `self.watsonx_model_gateway` or `self.watsonx_model` is not None.
But mypy can't detect that.